cmake/config_generation.cmake

```diff
@@ -54,65 +53,73 @@ set(_tbbbind_bin_version ${tbb_gen_cfg_TBBBIND_BINARY_VERSION})
     if (tbb_gen_cfg_SYSTEM_NAME STREQUAL "Linux")
         set(TBB_LIB_PREFIX "lib")
         set(TBB_LIB_EXT "so.\${_\${_tbb_component}_bin_version}")
-        set(TBB_IMPLIB_RELEASE "")
-        set(TBB_IMPLIB_DEBUG "")
-        if (tbb_gen_cfg_HANDLE_SUBDIRS)
-            set(TBB_HANDLE_SUBDIRS "
-if (CMAKE_SIZEOF_VOID_P STREQUAL \"8\")
-    set(_tbb_subdir intel64/gcc4.8)
-else ()
-    set(_tbb_subdir ia32/gcc4.8)
-endif()
+
+        set (TBB_HANDLE_IMPLIB "
+            set (_tbb_release_dll \${_tbb_release_lib})
+            set (_tbb_debug_dll \${_tbb_debug_lib})
 ")
+        if (tbb_gen_cfg_HANDLE_SUBDIRS)
+            set(TBB_HANDLE_SUBDIRS "set(_tbb_subdir gcc4.8)")
 
-            set(_libdir_for_pc_file "\${prefix}/lib/intel64/gcc4.8")
+            set(_libdir_for_pc_file "\${prefix}/lib/intel64/gcc4.8;\${prefix}/lib/")
             configure_file(${_tbb_gen_cfg_path}/../integration/pkg-config/tbb.pc.in ${config_install_dir}/tbb.pc @ONLY)
 
-            set(_libdir_for_pc_file "\${prefix}/lib/ia32/gcc4.8")
+            set(_libdir_for_pc_file "\${prefix}/lib/ia32/gcc4.8;\${prefix}/lib32/")
             configure_file(${_tbb_gen_cfg_path}/../integration/pkg-config/tbb.pc.in ${config_install_dir}/tbb32.pc @ONLY)
         endif()
     elseif (tbb_gen_cfg_SYSTEM_NAME STREQUAL "Darwin")
         set(TBB_LIB_PREFIX "lib")
         set(TBB_LIB_EXT "\${_\${_tbb_component}_bin_version}.dylib")
-        set(TBB_IMPLIB_RELEASE "")
-        set(TBB_IMPLIB_DEBUG "")
+
+        set (TBB_HANDLE_IMPLIB "
+            set (_tbb_release_dll \${_tbb_release_lib})
+            set (_tbb_debug_dll \${_tbb_debug_lib})
+")
         set(_libdir_for_pc_file "\${prefix}/lib")
         configure_file(${_tbb_gen_cfg_path}/../integration/pkg-config/tbb.pc.in ${config_install_dir}/tbb.pc @ONLY)
     elseif (tbb_gen_cfg_SYSTEM_NAME STREQUAL "Windows")
         set(TBB_LIB_PREFIX "")
-        set(TBB_LIB_EXT "dll")
+        set(TBB_LIB_EXT "lib")
         set(TBB_COMPILE_DEFINITIONS "
                                   INTERFACE_COMPILE_DEFINITIONS \"__TBB_NO_IMPLICIT_LINKAGE=1\"")
 
         # .lib files installed to TBB_LIB_REL_PATH (e.g. <prefix>/lib);
         # .dll files installed to TBB_DLL_REL_PATH (e.g. <prefix>/bin);
         # Expand TBB_LIB_REL_PATH here in IMPORTED_IMPLIB property and
         # redefine it with TBB_DLL_REL_PATH value to properly fill IMPORTED_LOCATION property in TBBConfig.cmake.in template.
-        set(TBB_IMPLIB_RELEASE "
-                                      IMPORTED_IMPLIB_RELEASE \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}.lib\"")
-        set(TBB_IMPLIB_DEBUG "
-                                      IMPORTED_IMPLIB_DEBUG \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}_debug.lib\"")
-        set(TBB_LIB_REL_PATH ${TBB_DLL_REL_PATH})
+        set (TBB_HANDLE_IMPLIB "
+            find_file(_tbb_release_dll
```

Comment:
why do you need to generate all this stuff while cmake can do it for you?
Just install targets to export list `TBBTargets`, install this config using:
```

install(EXPORT TBBTargets
        FILE TBBTargets.cmake
        NAMESPACE TBB::
        ...
```

Reply:
It is actually already there https://github.com/oneapi-src/oneTBB/blob/master/CMakeLists.txt#L259
```CMake
install(EXPORT ${PROJECT_NAME}Targets
        NAMESPACE TBB::
        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
        COMPONENT devel)
```
This `config_generation` thing is used outside of the project's CMake scope. The idea is to be able to work with multiple packages' folder structure like oneTBB GitHub package (for example, Windows libraries are located under `<tbbroot>/lib/vc14 or vc14_uwp`, Linux libraries are located under `<tbbroot>/lib/gcc4.8`) or Conda (for both Windows and Linux it is just `lib/`). And it is easier to test it since you have one config file.

I am not a CMake pro in terms of installation and packaging so I deal with the approach that was introduced before :)
If you have an idea on how to do the thing described above in simpler way I would appreciate if you'd share it.